client/components/videos-ui/index.jsx

```diff
@@ -50,13 +50,26 @@ const VideosUi = ( { headerBar, footerBar } ) => {
 	} );
 
 	useEffect( () => {
-		if ( ! currentVideoKey && course ) {
-			// @TODO add logic to pick the first unseen video
+		if ( ! course ) {
+			return;
+		}
+
+		const videoSlugs = Object.keys( course.videos );
+		if ( ! currentVideoKey ) {
 			const initialVideoId = 'find-theme';
 			setCurrentVideoKey( initialVideoId );
-			setSelectedVideoIndex( Object.keys( course.videos ).indexOf( initialVideoId ) );
+			setSelectedVideoIndex( videoSlugs.indexOf( initialVideoId ) );
 		}
-	}, [ currentVideoKey, course ] );
+
+		const viewedSlugs = Object.keys( userCourseProgression );
+		if ( viewedSlugs.length > 0 ) {
+			const nextSlug = videoSlugs.find( ( x ) => ! viewedSlugs.includes( x ) );
```

Comment:
I think you misunderstand me here; it's the `x` variable I didn't like. :)

Reply:
🤦🏻‍♂️, I'll fix that